src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,64 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static int AGE_AT_LEAST_REQUIRE = 18;
+    private static int LENGTH_AT_LEAST_REQUIRE = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNull(user);
+        checkAge(user.getAge());
+        checkLogin(user.getLogin());
+        checkPassword(user.getPassword());
+        storageDao.add(user);
+        return user;
+    }
+
+    private void checkAge(Integer age) {
+        if (age < AGE_AT_LEAST_REQUIRE) {
+            try {
+                throw new RegistrationException("Not valid login");
+            } catch (RegistrationException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
🤯 
You shouldn't catch a registration exception and rethrow a more general one. What's the point of this custom exception then? just throw, it should be a runtime one, no checks/catches are needed

Reply:
\+

Reply:
\+